extensions/ql-vscode/src/vscode-tests/cli-integration/remote-queries/variant-analysis-manager.test.ts

```diff
@@ -359,409 +359,418 @@ describe("Variant Analysis Manager", () => {
+                [scannedRepos[0].repository.id]: {
+                  repositoryId: scannedRepos[0].repository.id,
+                  downloadStatus:
+                    VariantAnalysisScannedRepositoryDownloadStatus.Failed,
+                },
+                [scannedRepos[1].repository.id]: {
+                  repositoryId: scannedRepos[1].repository.id,
+                  downloadStatus:
+                    VariantAnalysisScannedRepositoryDownloadStatus.Succeeded,
+                },
               },
-              [scannedRepos[2].repository.id]: {
-                repositoryId: scannedRepos[2].repository.id,
-                downloadStatus:
-                  VariantAnalysisScannedRepositoryDownloadStatus.InProgress,
+            );
+          });
+
+          it("should have a failed repo state when the download fails", async () => {
+            getVariantAnalysisRepoResultStub.mockRejectedValueOnce(
+              new Error("Failed to download"),
+            );
+
+            await expect(
+              variantAnalysisManager.autoDownloadVariantAnalysisResult(
+                scannedRepos[0],
+                variantAnalysis,
+                cancellationTokenSource.token,
+              ),
+            ).rejects.toThrow();
+
+            expect(outputJsonStub).not.toHaveBeenCalled();
+
+            await variantAnalysisManager.autoDownloadVariantAnalysisResult(
+              scannedRepos[1],
+              variantAnalysis,
+              cancellationTokenSource.token,
+            );
+
+            expect(outputJsonStub).toHaveBeenCalledWith(
+              join(
+                storagePath,
+                variantAnalysis.id.toString(),
+                "repo_states.json",
+              ),
+              {
+                [scannedRepos[0].repository.id]: {
+                  repositoryId: scannedRepos[0].repository.id,
+                  downloadStatus:
+                    VariantAnalysisScannedRepositoryDownloadStatus.Failed,
+                },
+                [scannedRepos[1].repository.id]: {
+                  repositoryId: scannedRepos[1].repository.id,
+                  downloadStatus:
+                    VariantAnalysisScannedRepositoryDownloadStatus.Succeeded,
+                },
               },
-              [scannedRepos[0].repository.id]: {
-                repositoryId: scannedRepos[0].repository.id,
-                downloadStatus:
-                  VariantAnalysisScannedRepositoryDownloadStatus.Succeeded,
+            );
+          });
+
+          it("should update the repo state correctly", async () => {
+            // To set some initial repo states, we need to mock the correct methods so that the repo states are read in.
+            // The actual tests for these are in rehydrateVariantAnalysis, so we can just mock them here and test that
+            // the methods are called.
+
+            pathExistsStub.mockImplementation(() => true);
+            // This will read in the correct repo states
```

Comment:
Could we drop this comment and instead wrap the code it's describing in a method? E.g. `loadRepoStates(readJsonStub, scannedRepos)`